vortex-array/src/vtable/statistics.rs

```diff
@@ -22,3 +23,35 @@ where
         StatisticsVTable::compute_statistics(encoding, array_ref, stat)
     }
 }
+
+pub fn compute_statistics(array: &Array, stat: Stat) -> VortexResult<StatsSet> {
+    if array.is_empty() {
+        return Ok(StatsSet::empty_array());
+    }
+
+    let mut set = if stat == Stat::Min || stat == Stat::Max {
+        // min max sets the array stats
+        let _min_max = min_max(array)?;
+        array.to_set()
+    } else {
+        array.vtable().compute_statistics(array, stat)?
```

Comment:
Is the idea that all stats end up as compute functions?

Reply:
Yep, also we can share further stat inference logc.